java/runtime/src/main/java/io/ray/runtime/object/ObjectStore.java

```diff
@@ -181,4 +186,19 @@ public abstract void delete(List<ObjectId> objectIds, boolean localOnly,
    * @param objectId The object ID to decrease the reference count for.
    */
   public abstract void removeLocalReference(UniqueId workerId, ObjectId objectId);
+
+  /**
+   * Promote the given object to the underlying object store.
+   * @param objectId The ID of the object to promote
+   * @return the serialized ownership address
+   */
+  public abstract byte[] promoteAndGetOwnershipInfo(ObjectId objectId);
+
+  /**
+   * Register the given object ownership information.
+   * @param outer The outer ObjectId which contains the given ObjectId
+   * @param objectId The ID of the object needs to register
+   * @param ownership The serialized address of the owner
+   */
+  public abstract void registerObjectRef(ObjectId outer, ObjectId objectId, byte[] ownership);
```

Comment:
Can you align it with the C++ declaration? e.g. the order of parameters, `outer` -> `outerObjectId`, `ownership` -> `ownerAddress`.

https://github.com/ray-project/ray/blob/427ba3df47f826f5114c81a98ff29ba1fc8efd13/src/ray/core_worker/core_worker.h#L444

Reply:
updated